components/script/dom/htmltableelement.rs

```diff
@@ -395,19 +395,33 @@ impl HTMLTableElementMethods for HTMLTableElement {
         Ok(new_row)
     }
 
-    // https://html.spec.whatwg.org/multipage/#dom-table-deleterow
+    /// <https://html.spec.whatwg.org/multipage/#dom-table-deleterow>
     fn DeleteRow(&self, mut index: i32) -> Fallible<()> {
         let rows = self.Rows();
-        // Step 1.
-        if index == -1 {
-            index = rows.Length() as i32 - 1;
-        }
-        // Step 2.
-        if index < 0 || index as u32 >= rows.Length() {
+
+        // Step 1: If index is less than −1 or greater than or equal to the number of elements
+        // in the rows collection, then throw an "IndexSizeError".
+        if index < -1 || index >= rows.Length() as i32 {
             return Err(Error::IndexSize);
         }
-        // Step 3.
-        DomRoot::upcast::<Node>(rows.Item(index as u32).unwrap()).remove_self();
+
+        let num_rows = rows.Length() as i32;
+
+        // Step 2: If index is −1, then remove the last element in the rows collection from its
+        // parent, or do nothing if the rows collection is empty.
+        if (-1..num_rows).contains(&index) {
```

Comment:
I think it makes sense to move this check above  to where you return `Err` and reverse it, as it's simply another way of saying the reverse of "if index < -1 || index >= rows.Length() as i32 {" then you can de-indent the code in this block.

Reply:
Ah thank you - I think I misinterpreted your earlier change request. That's been updated 👍